Sources/HummingbirdHTTP2/HTTP2Channel.swift

```diff
@@ -0,0 +1,97 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Hummingbird server framework project
+//
+// Copyright (c) 2023 the Hummingbird authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See hummingbird/CONTRIBUTORS.txt for the list of Hummingbird authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Logging
+import NIOCore
+import NIOPosix
+import NIOHTTP1
+import NIOHTTP2
+import Hummingbird
+import HummingbirdCore
+import NIOSSL
+
+public struct HTTP2Channel: HTTPChannelHandler {
+    public typealias Value = EventLoopFuture<NIONegotiatedHTTPVersion<HTTP1Channel.Value, (NIOAsyncChannel<HTTP2Frame, HTTP2Frame>, NIOHTTP2Handler.AsyncStreamMultiplexer<HTTP1Channel.Value>)>>
+
+    private var tlsConfiguration: TLSConfiguration
+    private var http1: HTTP1Channel
+    public var responder: @Sendable (HBHTTPRequest, Channel) async throws -> HBHTTPResponse {
+        get { http1.responder }
+        set { http1.responder = newValue }
+    }
+
+    public init(
+        tlsConfiguration: TLSConfiguration,
+        http1: HTTP1Channel = HTTP1Channel(),
+        responder: @escaping @Sendable (HBHTTPRequest, Channel) async throws -> HBHTTPResponse = { _, _ in throw HBHTTPError(.notImplemented) }
+    ) {
+        self.tlsConfiguration = tlsConfiguration
+        self.http1 = http1
+        // self.additionalChannelHandlers = additionalChannelHandlers
+        self.responder = responder
+    }
+
+    public func initialize(channel: Channel, configuration: HBServerConfiguration, logger: Logger) -> EventLoopFuture<Value> {
+        channel.eventLoop.flatSubmit {
+            do {
+                let sslContext = try NIOSSLContext(configuration: tlsConfiguration)
```

Comment:
Can we create this once in the initializer and store it as a member variable instead of the TLSConfiguration. Means we avoid creating it for every new child channel.

Reply:
I kinda assumed we couldn't share it across channels or even eventloops

Reply:
Its docs clearly prove me wrong, so I've been mishandling that for a while now